src/web/utils/apiError.tsx

```diff
@@ -25,3 +27,18 @@ export function backendError(e: unknown, overrideMessage: string) {
   }
   return Error(overrideMessage);
 }
+
+export function handleErrorPopup(
+  e: Error,
+  setStatusPopup: React.Dispatch<React.SetStateAction<StatusNotificationType | undefined>>,
+  setShowStatusPopup: React.Dispatch<React.SetStateAction<boolean>>
+) {
+  const hasHash = Object.hasOwn(e, 'errorHash') && (e as ApiError).errorHash;
+  const hash = hasHash ? `: (${(e as ApiError).errorHash})` : '';
+  setStatusPopup({
+    type: 'Error',
+    message: `${e.message}${hash}`,
```

Comment:
We don't know that `e` has a `.message` - it just looks like we do because of the `as` in the catch blocks - so in some cases `.message` might be undefined.

Reply:
good point - I've added a type check at the top of the method.